CMakeLists.txt

```diff
@@ -292,6 +292,10 @@ if (NOT USE_SIMD STREQUAL "")
             else ()
                 add_definitions ("-m${feature}")
             endif ()
+            if (feature STREQUAL "fma" AND CMAKE_COMPILER_IS_CLANG)
+                # Clang needs an explicit flag to make fma ops
+                add_definitions ("-ffp-contract=fast")
```

Comment:
I would be fine with upgrading all uses of `madd` (i.e.: I don't see a need to make a seperate `madd_aggressive`), because we already use it only in specific parts of OIIO.

It just comes down to the risk vs. reward of turning on FMA blindly in users's code (for example what it does to `lerp`).

From what I saw on the `clang` mailing list, it seemed like there was hesitation to follow GCC's default. To be fair, the standard makes little guarantees with any of this - so anything is fair game from the compiler side.

My main worry is that once `fma` is on, if the compiler is allowed to swap it in anywhere it likes (or not) you could end up having subtle behavioral differences between opt and debug, or even between opt builds based on code movement around the expressions. It some ways it ends up bringing back the undesirable behavior of the old x87 FPU that had mixed precision depending upon what was written back to memory.

All that being said -- I still very much want to take advantage of FMA where possible ... I just wish there was a way to ''guarantee'' where it gets applied.